### ISSUE TITLE
Applet recreates "Radio3.0" directory at default location upon login

### ISSUE THREAD

**Issue by Phx01** [linuxmint/cinnamon-spices-applets#5527]
### Applet version/Build date

1.61.1

### Cinnamon version

6.0.4

### Distribution

Fedora 39

### Graphics hardware *and* driver used

Intel Celeron N5100 [UHD Graphics] / i915

### Applet name and maintainer

Radio3.0@claudiux @claudiux

### What happened?

Despite relocating the "Radio3.0" directory to another sub-directory of my home directory, upon logging in to my user the directory is still re-generated at its default location directly in `~/` readily with radio icon.

#### Comparison to my primary computer
My main workstation had this plugin installed quite some time before and despite having updated it to the same version does not exploit this behavior there.

#### Issue start
While I am not fully certain, if I recall right this behavior started with v1.15.x.

#### What I tried

- I removed and then reinstalled this applet.
- I reset the directory to the default location and then set it up again in the desired location

#### Furthermore
I gladly provide further details once I know what is needed.

**Comment by claudiux** [comment]
@Phx01 
Hi,
Thanks for reporting this issue.
This is quite strange because I can't reproduce it on my Linux Mint 21.3.
I'll try on Fedora 39.

**Comment by claudiux** [comment]
@Phx01 The old directory has not been deleted because it is not empty. Please move yourself the files from the old directory to the new one; then delete the old directory.
In settings, select the path to the new directory.
Please tell me if that works.

**Comment by Phx01** [comment]
Each time I log in I manually delete the directory.  So, it cannot contain anything really unless the applet fills it with something upon creation.  However, I could not find any visible files (tested using `ls -al`).  Following its manual removal, the directory stays away for the duration of the session.  It will be re-created, however, once I log out and back in (or restart the computer).

Since I run Nextcloud in a mobile profile kind of setup, I can see the creation taking place on the main machine just seconds after I log in to the secondary computer since the directory is synced back to my workstation (albeit without the icon).  Removing it from either side repeats the above-mentioned cycle of it staying away for the remainder of the session.

I second that this is odd and difficult to reproduce since I cannot even see it happening on my workstation using more or less the same setup.  File system-wise, the differences are

- the secondary machine uses BTRFS while the workstation uses XFS
- the secondary machine uses LUKS encryption while the workstation does not use any block-level encryption

Though either difference should not invoke such a behavior on a user level.  So, I am bewildered as to why this happens.

If it helps, I can see about uninstalling it from the secondary machine once again, remove all remainders and then re-install it again.  For that I will only need to know all locations which are used to store information about this applet, so I can make it "never been there".  A regular uninstall apparently leaves some traces somewhere either readily from the applet or somewhere in Cinnamon Desktop.

**Comment by claudiux** [comment]
Thank you for all these details. We'll try to check the same things on each of your computers:

- Where is the file Radio3.0@claudiux.json? In `~/.config/cinnamon/spices/Radio3.0@claudiux` or in `~/.cinnamon/configs/Radio3.0@claudiux`?

This file contains the string "recording-path" twice, so please copy the 6 lines that follow the second time here.

It should look like this:

```
"recording-path": {
        "type": "filechooser",
        "description": "Path to the folder that will contain your future recordings",
        "select-dir": true,
        "default": "",
        "value": "file://PATH/TO/Radio3.0"
    },
```

**Comment by Phx01** [comment]
The file `Radio3.0@claudiux.json` is located inside `~/.config/cinnamon/spices/Radio3.0@claudiux`.  Considering the other possible path, this means that I downloaded it through the "Cinnamon Settings" » "Applets" option (which I did).

The "recording-path" array looks as follows:
```
"recording-path": {
        "type": "filechooser",
        "description": "Path to the folder that will contain your future recordings",
        "select-dir": true,
        "default": "",
        "value": "file:///Users/ipa/_MYUSER_/Media/Radio3.0"
    },
```
The only thing I redacted was my username and replaced it with `_MYUSER_`.  The relocated directory, `~/Media/Radio3.0`, exists.

**Comment by claudiux** [comment]
Your `Radio3.0@claudiux.json` is at the right place for Cinnamon 6.0.

This result is well the same on both computers, isn't it?

`/Users/ipa/_MYUSER_` is it well mounted **before** this applet starts? 

Please try this:

  cd ~/Media/Radio3.0
  pwd

Then replace in this file the "value" field by `file://` followed by the result of this pwd command. Save the file then restart Cinnamon.

Tell me if that works.

**Comment by Phx01** [comment]
I made the modifications on both computers.  Though it is worth noting that it the mentioned line is on line 294 (however, on both computers).

The main workstation shows it as follows:
```
Gjs-Message: 20:12:55.328: JS LOG: [LookingGlass/info] Reloading applet: Radio3.0@claudiux/25
Gjs-Message: 20:12:55.589: JS LOG: [LookingGlass/info] USER_MUSIC_DIR: /Users/ipa/_MYUSER_/Media/Music
Gjs-Message: 20:12:55.665: JS LOG: [LookingGlass/info] Loaded applet Radio3.0@claudiux in 159 ms
```

The secondary computer shows the lines like this:
```
Gjs-Message: 20:12:36.249: JS LOG: [LookingGlass/info] Reloading applet: Radio3.0@claudiux/22
Gjs-Message: 20:12:36.449: JS LOG: [LookingGlass/info] USER_MUSIC_DIR: /Users/ipa/_MYUSER_
Gjs-Message: 20:12:36.547: JS LOG: [LookingGlass/info] Loaded applet Radio3.0@claudiux in 191 ms
```

While writing this to you, I noted that the first line differs as well with the workstation noting `Radio3.0@claudiux/25` and the secondary computer having it as `Radio3.0@claudiux/22`.  Though clicking on the applet either computer reports the same version: v1.16.1.  Not knowing much about how applets work in Cinnamon, this could as well be the loading order in Cinnamon itself.

**Comment by Phx01** [comment]
Ok.  The output made me think.  I looked at the recent modification we did to `applet.js` and traced it back to `UserDirectory.DIRECTORY_MUSIC`.  I then opened `~/.config/user-dirs.dirs` and found the solution:

I created the `Media` directory in my home and moved the media directories (Music, Pictures & Videos) into it.  I also took care of updating `~/.config/user-dirs.dirs` for it.  However, for some twist of fate, these modifications were reverted, now only showing `$HOME/`.

I now updated it again through `ssh` and then logged in to the GUI on the secondary machine and the problem disappeared.  With that discovery I feel embarrassed to have used up so much time for you.

At any rate:  Thanks!  With your help I discovered the error and will close this issue.  Verdict: User error.

**Comment by Phx01** [comment]
Actually, I need to re-open this one.  While the original issue (the difference between the workstation and secondary computer) is now solved, with the recent discovery of the origin, I still find that the directory is created inside the user's Music directory even though I configured it to be inside the `~/Media` directory.  That is, this time it is consistent on both machines.

In the applet I changed the recording directory to `~/Media/Radio3.0`.  Since I created the directory there, it also exists.  However, there is a directory `~/Media/Music/Radio3.0` (this one also uses the nice icon), which should not be there (since `~/Media/Music` is monitored by Rhythmbox and it shall not pick up anything in the Radio directory).

So, somehow the user configuration in the applet is not (fully) regarded.  I never recorded anything, so I do not know if the custom-set directory is actually used or if the directory using the icon is used instead (my bet is almost to the latter since it uses the radio icon).

**Comment by claudiux** [comment]
OK, I'll make sure this directory is not created by the applet. It's up to the user to create it and indicate the correct path. As long as this directory does not exist, no recording will be possible. This will solve the problem once and for all. Unless, of course, the configurations need to be identical on two computers with different configurations.

**Comment by Phx01** [comment]
I think that this may be a bit too drastic.  When installing the applet as is, it has a default directory configured and that is fine to be created by the applet upon start.  However, once the user sets a different location, then the new location should be "monitored" instead of the default location.

Example (expected behavior):
- if I freshly install the applet, the default directory in `UserDirectory.DIRECTORY_MUSIC` is just fine
- if I remove the directory, it is also fine to be recreated at that location
- if I go to the applet's settings and configure a new location (e.g. in my case `~/Media`), then the directory should be (re-)created at the new location and the default location disregarded

I can also think of an additional setting to choose 'none' as a directory for recordings, but this would then be a feature request rather than a bug.

**Comment by claudiux** [comment]
@Phx01 This problem seems to have been solved with version 1.17.3 of `Radio3.0@claudiux`. Can you please confirm?

**Comment by Phx01** [comment]
Hi @claudiux !  I almost thought you were too annoyed by me with the previous response.  My apologies if I did.

Yes, I can confirm it being fixed (in the meantime being v1.17.5).  Thank you so much!  I will close this bug report.

**Comment by claudiux** [comment]
@Phx01 Anything that can help me improve this applet is welcome. Thank you again for reporting this problem. I'm glad you're satisfied with the new version.
